app/scripts/controllers/metametrics-controller.ts

```diff
@@ -1135,6 +1135,23 @@ export default class MetaMetricsController extends BaseController<
     }
     ///: END:ONLY_INCLUDE_IF
 
+    let chainId;
+    if (
+      properties &&
+      'chain_id' in properties &&
+      typeof properties.chain_id === 'string'
+    ) {
+      chainId = properties.chain_id;
+    } else if (
+      properties &&
+      'chain_id_caip' in properties &&
+      typeof properties.chain_id_caip === 'string'
+    ) {
```

Comment:
should this guard come before the first guard? otherwise it's still possible for both `chain_id` and `chain_id_caip` to be defined but for chainId to end up not being null

Reply:
quite right!

Reply:
fixed here: https://github.com/MetaMask/metamask-extension/pull/32586/commits/c4286d0c9914a106168ac079f47f6787618f437f